src/openapi/streaming/serializer/serializer-interface.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Serialization Interface
+ * @constructor
+ */
+function SerializerInterface(name, engine = null) {

```

Comment:
unusual to have implementation in an "interface"? same with getFormatName below? I think both implementations do their own thing anyway and don't use these?

Reply:
Yeah, Interface wasn't best name. I just hate suffixing/prefixing it with 'Base', and Serializer alone seems vague. 

Will change it to Base however, is if we would like to follow interface requirements, We should implement all methods.

Reply:
Base makes most sense to me personally if it contains implementation.
Contract sounds like interface to me, I wouldn't expect any implementation. I'm not particularly tied to either though, as long as it's not interface :)